### DISCUSSION TITLE
How to build googletest for RISC-V

### DISCUSSION THREAD

**Discussion by alyona-eryomina** [google/googletest#4119] [Community Help]
Hi! I want to cross-build googletest for RISC-V. I have a RISC-V toolchain for cross-compiling, so I want to build on x86, but for RISC-V. How can I do that? I cannot find any guides about cross-building

**Comment by Mahiuha** [comment]
To cross-build Google Test for RISC-V on an x86 machine, you will need to use the RISC-V toolchain that you have. The process will involve configuring the build system to use the RISC-V toolchain, and then building the library.
Here are the general steps you can follow:

1. Clone the Google Test repository from GitHub: git clone https://github.com/google/googletest.git
2. Navigate to the googletest directory: cd googletest
3. Create a build directory: mkdir build
4. Navigate to the build directory: cd build
5. Run CMake to configure the build, specifying the RISC-V toolchain and specifying the target architecture, for example:
    ```
         cmake .. -DCMAKE_C_COMPILER=riscv64-unknown-elf-gcc \
         -DCMAKE_CXX_COMPILER=riscv64-unknown-elf-g++ \
         -DCMAKE_SYSTEM_NAME=Generic \
         -DCMAKE_SYSTEM_PROCESSOR=riscv
   ```
6. Build the library: make

Note that this is a general guide, and the specific steps may vary depending on the version of Google Test you are using and the configuration of your RISC-V toolchain.
Also, you may need to adjust the paths to the toolchain and change other configurations.
